### PR TITLE
Improve Kafka docs to configure authentication

### PR DESCRIPTION
**Pull request by juan131**
This PR improves the Kafka docs to configure authentication based on the improvements that will be added to the chart to support both PEM and JKS formats to configure TLS.

### CODE REVIEW

File: charts/kafka/_enable_tls.md.erb [hunk 1]
```diff
@@ -9,37 +9,80 @@ weight: 20
...
+auth.tls.type=pem
+auth.tls.existingSecret=kafka-tls
+auth.tls.password=some-password
+~~~
+
+### Use self-signed TLS certificates
+
+In case you want to use self-signed TLS certificates using the PEM format, set the *auth.tls.autoGenerated* parameter to *true*.
+
+To deploy a Kafka cluster with 3 Kafka brokers and TLS authenticationboth for inter-broker and client communications, use the following parameters:
```
Comment (vikram-bitnami) [inline charts/kafka/_enable_tls.md.erb:72]: authenticationboth - needs space

File: charts/kafka/_enable_tls.md.erb [hunk 2]
```diff
@@ -9,37 +9,80 @@ weight: 20
...
+replicaCount=3
+auth.clientProtocol=tls
+auth.interBrokerProtocol=tls
+auth.tls.type=pem
+auth.tls.autoGenerated=true
+~~~
+
+### Expose metrics when TLS authentication is enabled
+
+If you also enable exposing metrics using the Kafka expoter, and you are using *sasl_tls*, *tls*, or *mtls* authentication protocols for client connections, you need to mount the CA certificated used to sign the brokers certificates in the exporter so it can validate the Kafka brokers. To do so, create a secret containing the CA, and set the *metrics.kafka.certificatesSecret* parameter. As an alternative, you can skip TLS validation using extra flags:
```
Comment (vikram-bitnami) [inline charts/kafka/_enable_tls.md.erb:84]: certificated -> typo
expoter -> typo

If you plan to expose metrics using the Kafka exporter, and you are using *sasl_tls*, *tls*, or *mtls* authentication protocols for client connections, the exporter must have the ability to validate the Kafka brokers. This is done by mounting the CA certificate, used to sign the broker certificates, in the exporter as a secret, and then setting the *metrics.kafka.certificatesSecret* parameter to point to this secret. As an alternative, you can skip TLS validation using extra flags, as shown below:

Comment (vikram-bitnami) [reply charts/kafka/_enable_tls.md.erb:84]: for this `To do so, create a secret containing the CA, and set the *metrics.kafka.certificatesSecret* parameter.` do you want to add example code?

File: charts/kafka/_enable_tls.md.erb [hunk 3]
```diff
@@ -9,37 +9,80 @@ weight: 20
...
+In the first two cases, you will have to create the TLS certificates below in advance.
+
+* If you are using the JKS format, a truststore (*kafka.truststore.jks*) and one keystore (*kafka.keystore.jks*) per Kafka broker in the cluster.
+* If you are using the PEM format, a CA (*kafka.truststore.pem*) and one public certificate and private key (*kafka.keystore.pem* and *kafka.keystore.key*) per Kafka broker in the cluster.
+
+> NOTE: If the JKS keystore (JKS format) or the private key (PEM format) are password protected (recommended), you will need to provide the password to get access to them. To do so, use the *auth.tls.password* parameter to provide your password.
+
+#### Use JKS format
+
+To configure TLS authentication with JKS format on a Kafka cluster with 2 Kafka brokers use the command below to create the secret containing the TLS certificates:
```
[the remark this answers is not in the document]
Comment (vikram-bitnami) [reply charts/kafka/_enable_tls.md.erb:36]: two

File: charts/kafka/_enable_tls.md.erb [hunk 4]
```diff
@@ -9,37 +9,80 @@ weight: 20
...
+
+    $ kubectl create secret generic kafka-tls --from-file=./kafka.truststore.jks --from-file=./kafka-0.keystore.jks --from-file=./kafka-1.keystore.jks
+
+> NOTE: The command above assumes you already created the trustore and keystores files. [This script](https://raw.githubusercontent.com/confluentinc/confluent-platform-security-tools/master/kafka-generate-ssl.sh) can help you with the JKS files generation.
```
Comment (vikram-bitnami) [inline charts/kafka/_enable_tls.md.erb:40]: trustore -> typo

keystore files

 A [script to help with JKS file generation] is also available. -> SEO improvement

File: charts/kafka/_enable_tls.md.erb [hunk 5]
```diff
@@ -9,37 +9,80 @@ weight: 20
...
-
-    $ metrics.kafka.extraFlags={tls.insecure-skip-tls-verify: ""}
+auth.tls.type=jks
+auth.tls.existingSecret=kafka-tls
+auth.tls.password=some-password
+~~~
+
+#### Use PEM format
+
+To configure TLS authentication with PEM format on a Kafka cluster with 2 Kafka brokers use the command below to create the secret containing the TLS certificates:
```
[the remark this answers is not in the document]
Comment (vikram-bitnami) [reply charts/kafka/_enable_tls.md.erb:54]: two

File: charts/kafka/_enable_tls.md.erb (hunk 1 above)
Comment (vikram-bitnami) [inline charts/kafka/_enable_tls.md.erb:72]: three

**Review by andresbono — APPROVED** [review]
LGTM

**Pull request merged**